bindings/src/main/java/io/pravega/storage/extendeds3/ExtendedS3SimpleStorageFactory.java

```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.storage.extendeds3;
+
+import com.emc.object.s3.S3Client;
+import com.emc.object.s3.S3Config;
+import com.emc.object.s3.jersey.S3JerseyClient;
+import io.pravega.segmentstore.storage.Storage;
+import io.pravega.segmentstore.storage.StorageFactory;
+import io.pravega.segmentstore.storage.chunklayer.ChunkedSegmentStorage;
+import io.pravega.segmentstore.storage.chunklayer.ChunkedSegmentStorageConfig;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+
+import java.util.concurrent.ExecutorService;
+
+/**
+ * Factory for ExtendedS3 {@link Storage} implemented using {@link ChunkedSegmentStorage} and {@link ExtendedS3ChunkStorage}.
+ */
+@RequiredArgsConstructor
+public class ExtendedS3SimpleStorageFactory implements StorageFactory {
```

Comment:
The StorageFactory is created during initialization sequence. This is current design and I did not want to make big changes to current initialization sequence and wanted the change to be as small as it could be. There is no good way to pass in the configurations to the factory without changing factory interface or otherwise make factory stateful (which is not a good idea).
That is the reason why i chose this path.

In next few iterations we'll remove the old code and at that time we can rename the factory etc.